src/components/Offers/New/form-components/PublicationEndDateComponent.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { Controller } from "react-hook-form";
+import PropTypes from "prop-types";
+import { KeyboardDatePicker } from "@material-ui/pickers";
+import { Tooltip } from "@material-ui/core";
+import { Info } from "@material-ui/icons";
+
+
+const PublicationEndDateComponent = ({ fields, disabled, errors, requestErrors, control }) => (
+    <>
+        <Controller
+            name="publishEndDate"
+            render={(
+                { field: { onChange, onBlur, name, value } },
+            ) => (
+                <KeyboardDatePicker
+                    margin="dense"
+                    value={value}
+                    label="Publication End Date"
+                    id="publishEndDate-input"
+                    name={name}
+                    disabled={disabled}
+                    onChange={(_, value) => {
+                        const date = new Date(value);
+                        date.setHours(23, 59, 59, 0);
```

Comment:
Can you add a comment to explain why this is necessary for the next developer looking at this?